lib/vtls/mbedtls.c

```diff
@@ -184,12 +186,17 @@ static Curl_send mbed_send;
 static CURLcode mbedtls_version_from_curl(int *mbedver, long version)
 {
   switch(version) {
+#if MBEDTLS_VERSION_MAJOR >= 3
+    case CURL_SSLVERSION_TLSv1_0:
+    case CURL_SSLVERSION_TLSv1_1:
+#else
```

Comment:
This makes the code too hard to parse. Since the TLS support is so vastly different, I think we should provide two different implementations of `mbedtls_version_from_curl` entirely and do the version preprocessing around those rather than pick this one apart.